src/fpsolve/functional.jl

```diff
@@ -40,31 +62,79 @@ function fpsolve!(integrator::DDEIntegrator, fpsolver::FPSolver{<:FPFunctionalCo
       end
     end
 
-    # complete interpolation data of DDE integrator for time interval [t, t+dt]
-    # and copy it to ODE integrator
-    # has to be done before updates to ODE integrator, otherwise history function
-    # is incorrect
-    if iscomposite(alg)
-      addsteps!(k, t, uprev, integrator.u, dt, f, p, cache.caches[integrator.cache.current],
-                false, true, true)
-    else
-      addsteps!(k, t, uprev, integrator.u, dt, f, p, cache, false, true, true)
-    end
-    @inbounds for i in 1:length(k)
-      copyat_or_push!(ode_integrator.k, i, k[i])
-    end
-
-    # update state of the dummy ODE solver
-    ode_integrator.u = integrator.u
-
     # check stopping criterion
     iter > 1 && (η = θ / (1 - θ))
     if η * ndu < κ && (iter > 1 || iszero(ndu))
       # fixed-point iteration converges
-      fpsolver.status = η < fpsolver.fast_convergence_cutoff ? FastConvergence : Convergence
+      fpsolver.status = η < fast_convergence_cutoff ? FastConvergence : Convergence
       fail_convergence = false
       break
     end
+
+    # perform Anderson acceleration
+    if fpcache isa FPAndersonConstantCache && iter < max_iter
```

Comment:
I did not check this block thoroughly but it looks quite similar to the nlsolve! one so I'm assuming this chunk is good.